### PR TITLE
UNDERTOW-1024 Add an experimental support for SameSite Cookie attribute

### PR DESCRIPTION
**Pull request by msfm**

### CODE REVIEW

**Comment by undertow-pull-request** [comment]
**Windows** [Build 1982](https://ci.wildfly.org/viewLog.html?buildId=51392&buildTypeId=UNDERTOW_MasterWindows) outcome was **FAILURE** using a merge of 710fd98beb25290c7bac3d7397cd481e90e37b39
Summary: Tests failed: 1, passed: 2760, ignored: 556 Build time: 00:33:40
### Failed tests
```
io.undertow.server.handlers.ReceiverTestCase.testAsyncReceivePartialBytes: java.lang.RuntimeException: test failed with i equal to 6
	at io.undertow.server.handlers.ReceiverTestCase.doTest(ReceiverTestCase.java:233)
	at io.undertow.server.handlers.ReceiverTestCase.testAsyncReceivePartialBytes(ReceiverTestCase.java:200)
Caused by: java.net.SocketTimeoutException: Read timed out
	at io.undertow.server.handlers.ReceiverTestCase.runTest(ReceiverTestCase.java:246)
	at io.undertow.server.handlers.ReceiverTestCase.doTest(ReceiverTestCase.java:231)
	at io.undertow.server.handlers.ReceiverTestCase.testAsyncReceivePartialBytes(ReceiverTestCase.java:200)


```

File: core/src/main/java/io/undertow/server/handlers/Cookie.java [hunk 1]
```diff
@@ -69,4 +69,12 @@
     String getComment();
 
     Cookie setComment(final String comment);
+
+    boolean isSameSite();
```
Comment (stuartwdouglas) [inline core/src/main/java/io/undertow/server/handlers/Cookie.java]: For compatibility reasons these should use JDK8 default methods.

**Comment by msfm** [comment]
Thanks for your comment. I've just updated this PR to use Java 8 default methods.

**Comment by undertow-pull-request** [comment]
**Windows** [Build 1984](https://ci.wildfly.org/viewLog.html?buildId=51553&buildTypeId=UNDERTOW_MasterWindows) outcome was **FAILURE** using a merge of 91dc7eef52d4b75f86b0d275b838c070f87698ed
Summary: Tests failed: 1, passed: 2760, ignored: 556 Build time: 00:30:47
### Failed tests
```
io.undertow.server.handlers.ReceiverTestCase.testAsyncReceivePartialBytes: java.lang.RuntimeException: test failed with i equal to 6
	at io.undertow.server.handlers.ReceiverTestCase.doTest(ReceiverTestCase.java:233)
	at io.undertow.server.handlers.ReceiverTestCase.testAsyncReceivePartialBytes(ReceiverTestCase.java:200)
Caused by: java.net.SocketTimeoutException: Read timed out
	at io.undertow.server.handlers.ReceiverTestCase.runTest(ReceiverTestCase.java:246)
	at io.undertow.server.handlers.ReceiverTestCase.doTest(ReceiverTestCase.java:231)
	at io.undertow.server.handlers.ReceiverTestCase.testAsyncReceivePartialBytes(ReceiverTestCase.java:200)


```

**Pull request merged**